client/src/components/Forms/GenericForm.vue

```diff
@@ -0,0 +1,292 @@
+        delete newHMModelFlat[this.model_key];
+        this.localHMModelFlat = Object.assign(
+          this.localHMModelFlat,
+          newHMModelFlat,
+        );
+        this.localModelChoicesData = newModelChoicesData;
+      },
+    );
+  },
+  watch: {
+    /**
+     * If the state is changed locally, pass it up to the parent.
+     */
+    localHMModelFlat: {
+      deep: true,
+      handler() {
+        this.$emit('onChange', clonedeep(this.localHMModelFlat));
+      },
+    },
+    /**
+     * If the state is changed by the parent, change the local values.
+     */
+    relevantHMModelFlat: {
+      deep: true,
+      handler(newHMModelFlat) {
+        if (!isEqual(this.localHMModelFlat, newHMModelFlat)) {
+          this.localHMModelFlat = clonedeep(this.relevantHMModelFlat);
+        }
+      },
+    },
+  },
+  methods: {
+    getDoubleFieldProps(parameterKey) {
+      const parameterProps = PARAMETER_PROPS[parameterKey];
+      const doubleProps = {
+        param: parameterKey,
+        range: false,
+      };
+
+      // If properties do not exist, pass the default values
+      if (!parameterProps) return doubleProps;
+
+      if (parameterProps.plainName) {
+        doubleProps.param = parameterProps.plainName;
+      }
+      if (parameterProps.html) {
+        doubleProps.html = parameterProps.html;
+      }
+      if (parameterProps.range) {
+        doubleProps.range = parameterProps.range;
+      }
+      if (parameterProps.min) {
+        doubleProps.min = parameterProps.min;
+      }
+      if (parameterProps.max) {
+        doubleProps.max = parameterProps.max;
+      }
+      return doubleProps;
+    },
+    /**
+     * Gets a label for the given parameter. This should only be used on
+     * parameters that have a string value.
+     */
+    getParameterLabel(parameterKey) {
+      const parameterProps = PARAMETER_PROPS[parameterKey];
+      if (parameterProps && parameterProps.plainName) {
+        return parameterProps.plainName;
+      }
+      return parameterKey;
+    },
+    /**
+     * Gets the options for a given parameter. This should only be used on
+     * parameters that have a string value.
+     */
+    getParameterOptions(parameterKey) {
+      const parameterProps = PARAMETER_PROPS[parameterKey];
+      if (!parameterProps || !parameterProps.options) {
+        console.error(`Error in parameter "${parameterKey}" \n`
+        + 'No options were provided for the string value parameter in '
+        + 'parameter_properties.js');
+        return {};
+      }
+      return parameterProps.options;
+    },
+    isVisible(parameterKey) {
+      if (PARAMETER_PROPS[parameterKey]
+        && PARAMETER_PROPS[parameterKey].visible === false
+      ) {
+        return false;
+      }
+      return true;
+    },
+    isSlider(parameterKey) {
+      const parameterProps = PARAMETER_PROPS[parameterKey];
+      if (!parameterProps || !parameterProps.rangeSlider) {
```

Comment:
```return !(!parameterProps || !parameterProps.rangeSlider)```

Reply:
isnt (!parameterProps || !parameterProps.rangeSlider) === !(parameterProps && parameterProps.rangeSlider)
this could be:
```js
return !(parameterProps && parameterProps.rangeSlider);

Reply:
Implemented with a slight variation to match correct logic. 😄